addons/sriov-device-plugin/enable

```diff
@@ -142,16 +142,19 @@ def _enable_sriovdp(resources: dict) -> None:
     while time.time() - now <= timeout:
         sriov_allocatable_resources = _get_sriov_allocatable_resources()
 
-        for resource, pci_addresses in resources.items():
-            if sriov_allocatable_resources[f"intel.com/{resource}"] != str(len(pci_addresses)):
+        try:
+            if all(
+                sriov_allocatable_resources[f"intel.com/{resource}"] == str(len(pci_addresses))
+                for resource, pci_addresses in resources.items()
+            ):
+                return
+            else:
                 print("SR-IOV Network Device Plugin not ready, " +
                     f"intel.com/{resource} does not have enough allocatable resources.")
-                break
-        else:
-            return
+        except KeyError:
```

Comment:
Instead `try`-ing and catching `KeyError` exceptions, it would far simpler to use:

```
sriov_allocatable_resources.get(f"intel.com/{resource}") == str(len(pci_addresses))
```

instead of:

```
sriov_allocatable_resources[f"intel.com/{resource}"] == str(len(pci_addresses))
```

The outcome is the same, but we'd no longer have to have the try-except block, making the code much cleaner. Could you update this?

Reply:
True. Replaced with `get`.